Source/ui/EditToneTab.cpp

```diff
@@ -93,8 +1360,276 @@ void EditToneTab::buttonStateChanged (juce::Button* button)
 
 void EditToneTab::comboBoxChanged (juce::ComboBox* button)
 {
-    if (button == &toneSelectionComboBox)
-    {
-        updateValues();
-    }
+    sendSysexPatchToneChange();
+}
+
+void EditToneTab::sendSysexPatchToneChange1(uint32_t address, uint8_t value)
+{
+      uint8_t data[5];
+      data[0] = 0x00; // address MSB
+      data[1] = 0x08; // address
+      data[2] = 0x28 + toneCount;  // address
+      data[3] = address & 127;  // address LSB
+      data[4] = value;                 // data
+      uint32_t checksum = 0;
+      for (size_t i = 0; i < 5; i++) {
+        checksum += data[i];
+        if (checksum >= 128) {
+          checksum -= 128;
+        }
+      }
+
+      uint8_t buf[12];
+      buf[0] = 0xf0;
+      buf[1] = 0x41;
+      buf[2] = 0x10; // unit number
+      buf[3] = 0x46;
+      buf[4] = 0x12; // command
+      checksum = 128 - checksum;
+      for (size_t i = 0; i < 5; i++) {
+        buf[i + 5] = data[i];
+      }
+      buf[10] = checksum;
+      buf[11] = 0xf7;
+
+      audioProcessor.mcuLock.enter();
+      audioProcessor.mcu->postMidiSC55(buf, 12);
+      audioProcessor.mcuLock.exit();
+}
+
+void EditToneTab::sendSysexPatchToneChange2(uint32_t address, uint8_t value)
+{
+      uint8_t data[6];
+      data[0] = 0x00; // address MSB
+      data[1] = 0x08; // address
+      data[2] = 0x28 + toneCount;  // address
+      data[3] = address & 127;  // address LSB
+      data[4] = (value & 0xf0) >> 4;                 // data
+      data[5] = value & 0x0f;                 // data
+      uint32_t checksum = 0;
+      for (size_t i = 0; i < 6; i++) {
+        checksum += data[i];
+        if (checksum >= 128) {
+          checksum -= 128;
+        }
+      }
+
+      uint8_t buf[13];
+      buf[0] = 0xf0;
+      buf[1] = 0x41;
+      buf[2] = 0x10; // unit number
+      buf[3] = 0x46;
+      buf[4] = 0x12; // command
+      checksum = 128 - checksum;
+      for (size_t i = 0; i < 6; i++) {
+        buf[i + 5] = data[i];
+      }
+      buf[11] = checksum;
+      buf[12] = 0xf7;
+
+      audioProcessor.mcuLock.enter();
+      audioProcessor.mcu->postMidiSC55(buf, 13);
+      audioProcessor.mcuLock.exit();
+}
+
+
+void EditToneTab::sendSysexPatchToneChange()
+{
+        sendSysexPatchToneChange2(1,(uint8_t(waveformSlider.getValue() - 1)));

```

Comment:
Does this send all the params every time something changes? Maybe there is a way to send only the ones that changed if that's the case

Reply:
I think it would be better if param changes were individual rather than batched like this. Especially with host automation this may become a problem.

Reply:
Fair point about automation, @mkruselj, though the likelihood of me rewriting it immediately is slim due to various commitments, not to mention, again, it would require a lot of conditional statements. Still, if youi don't beat me to revising, I'll see what I can do later. About to try to submit a separate pull request later as well.

Reply:
I might take a shot at it. Switch-case for the win here. Probably makes sense to put it in a separate file.